docs/rerank.md

```
@@ -0,0 +1,127 @@
+
+cache = FileCache(".cache/rerank")
+reranker = ListwiseReranker(
+    adapter, cache=cache, cache_namespace="azure:gpt-5.1"
+)
+```
+
+### Cost model
+
+For one query with pool size `N`, window size `W`, step `S`:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)